e2e/helpers.ts

```diff
@@ -113,20 +118,183 @@ export const uploadImage = async (
+  page,
+  sectionLocator,
+  sectionTitle,
+  buttonLocator
+}: {
+  page: Page
+  sectionLocator: Locator
+  buttonLocator: Locator
+  sectionTitle: string
+}) => {
+  await sectionLocator.getByText('Select...').click()
+  await sectionLocator.getByText(sectionTitle, { exact: true }).click()
+
+  await uploadImage(page, buttonLocator)
+}
+
+export const expectAddress = async (
+  locator: Locator,
+  address: { [key: string]: any },
+  isDeletion?: boolean
+) => {
+  const addressKeys = [
+    'country',
+
+    'state',
+    'province',
+
+    'district',
+
+    'village',
+    'town',
+    'city',
+
+    'residentialArea',
+    'addressLine1',
+
+    'street',
+    'addressLine2',
+
+    'number',
+    'addressLine3',
+
+    'postcodeOrZip',
+    'postalCode',
+    'zipCode'
+  ]
+
+  if (isArray(address.line)) {
+    address.addressLine1 = address.line[2]
+    address.addressLine2 = address.line[1]
+    address.addressLine3 = address.line[0]
+  }
+
+  const texts = addressKeys
+    .map((key) => address[key])
+    .filter((value) => Boolean(value))
+
+  if (isDeletion) {
+    const deletionLocators = await locator.getByRole('deletion').all()
+    for (let i = 0; i < texts.length; i++) {
+      await expect(deletionLocators[getDeletionPosition(i)]).toContainText(
+        texts[i]
+      )
+    }
+  } else await expectTexts(locator, texts)
+}
+
+/*
+  The deletion section is formatted like bellow:
+  	'-'
+    'Farajaland'
+    'Central'
+    'Ibombo'
+    ''
+    'Example Town'
+    'Mitali Residential Area'
+    '4/A'
+    '1324'
+
+*/
+const getDeletionPosition = (i: number) => i + (i < 3 ? 1 : 2)
+
+export const expectTexts = async (locator: Locator, texts: string[]) => {
+  for (const text of texts) {
+    await expect(locator).toContainText(text)
+  }
+}
+
+export const expectTextWithChangeLink = async (
+  locator: Locator,
+  texts: string[]
+) => {
+  expectTexts(locator, texts)
+  await expect(locator).toContainText('Change')
+}
+
 export const getLocationNameFromFhirId = async (fhirId: string) => {
   const res = await fetch(`${GATEWAY_HOST}/location/${fhirId}`)
   const location = (await res.json()) as fhir.Location
   return location.name
 }
 
-export async function continueForm(page: Page) {
+export async function continueForm(page: Page, label: string = 'Continue') {
+  /*
+   * This timeout is to ensure that all previous actions have been completed
+   * including filling inputs and that the changed values have been reflected
+   * also to the Redux state. 500ms is selected as a safe value.
+   */
+  await page.waitForTimeout(500)
+  return page.getByText(label, { exact: true }).click()
+}
+
+export async function goBackToReview(page: Page) {
   /*
    * This timeout is to ensure that all previous actions have been completed
    * including filling inputs and that the changed values have been reflected
    * also to the Redux state. 500ms is selected as a safe value.
    */
   await page.waitForTimeout(500)
```

Comment:
We could set this to global variable. This would allow us to change it in one place, if we try to reduce the timeout in the future, for example. And comment could be in a single place

```
export const SAFE_INPUT_CHANGE_TIMEOUT_MS = 500;
```